### PR TITLE
Protect fsScanner autosave settings w/ mutex

### PR DESCRIPTION
**Pull request by dsprenkels**
This PR adds a bit of bookkeeping to `Server.fsScanner`, to protect the autosaver goroutine settings from race conditions. I should note that this function (`Server.fsScanner`) is a good candidate for refactoring efforts.

Fixes issue #14.

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -222,28 +222,44 @@ func (s *Server) fsScanner(done <-chan void) {
...
+	autosaveStop := false
+	autosavePause := fastPass
+	var autosaveMtx sync.Mutex
 	go func() {
-		for range time.Tick(time.Minute) {
-			if firstPassDone {
-				return
+		// Returns true if goroutine should stop
+		autosaveFunc := func() bool {
+			autosaveMtx.Lock()
```
Comment (mrngm) [inline server.go]: It's unclear if this mutex also hold for `autosavePause`.

Comment (mrngm) [reply server.go]: Why not use RWMutex here? The only write action is at the end of the function.

Comment (dsprenkels) [reply server.go]: > It's unclear if this mutex also hold for `autosavePause`.

It does. Will comment accordingly.

File: server.go [hunk 2]
```diff
@@ -222,22 +222,36 @@ func (s *Server) fsScanner(done <-chan void) {
 		log.Printf("Couldn't load hashcache.dat: %v", err)
 	}
 	fastPass := (cacheSeed != nil)
-	firstPassDone := false
+
+	// Additionally autosave the hash cache each minute
+	autosaveStop := false
```
Comment (Jille) [inline server.go]: This'd be much easier with sync/atomic rather than mutexes btw. Or I guess channels would work well for this too.

**Review by Jille — COMMENTED** [review]
Yesterday you mentioned this was buggy, and while I agree it was a race, I did think through that the race wasn't problematic. Worst case it'd just take a little more time for the bools to propagate to the goroutine.

**Pull request merged**